packages/datatrak-web/src/features/Tasks/TasksTable/AssignTaskModal.tsx

```diff
@@ -61,24 +63,26 @@ export const AssignTaskModal = ({ task, Button }: AssignTaskModalProps) => {
         isLoading={isLoading}
       >
         <Container>
-          <form onSubmit={handleSubmit(editTask)}>
-            <Controller
-              name="assignee_id"
-              control={control}
-              rules={{ required: 'Required' }}
-              render={({ value, onChange, ref }, { invalid }) => (
-                <AssigneeInput
-                  value={value}
-                  required
-                  onChange={onChange}
-                  inputRef={ref}
-                  countryCode={task?.entity?.countryCode}
-                  surveyCode={task?.survey?.code}
-                  error={invalid}
-                />
-              )}
-            />
-          </form>
+          <FormProvider {...formContext}>
+            <TaskForm onSubmit={handleSubmit(editTask)}>
+              <Controller
+                name="assigneeId"
+                control={control}
+                rules={{ required: 'Required' }}
+                render={({ value, onChange, ref }, { invalid }) => (
+                  <AssigneeInput
+                    value={value}
+                    required
+                    onChange={onChange}
+                    inputRef={ref}
+                    countryCode={task?.entity?.countryCode}
```

Comment:
```suggestion
                    countryCode={task.entity.countryCode}
```
Might just be a matter of merging the other branch in but just mentioning again the optional chaining